src/FluentMigrator.Extensions.Postgres/Postgres/PostgresExtensions.IndexStorage.cs

```diff
@@ -0,0 +1,171 @@
+#region License
+// Copyright (c) 2020, FluentMigrator Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using FluentMigrator.Builders.Create.Index;
+using FluentMigrator.Infrastructure;
+using FluentMigrator.Infrastructure.Extensions;
+
+namespace FluentMigrator.Postgres
+{
+    public static partial class PostgresExtensions
+    {
+        public const string IndexFillFactor = "PostgresFillFactor";
+
+        /// <summary>
+        /// The fillfactor for an index is a percentage that determines how full the index method will try to pack index pages. For B-trees, leaf pages are filled to this percentage during initial index build,
+        /// and also when extending the index at the right (adding new largest key values). If pages subsequently become completely full, they will be split, leading to gradual degradation in the index's efficiency.
+        /// B-trees use a default fillfactor of 90, but any integer value from 10 to 100 can be selected. If the table is static then fillfactor 100 is best to minimize the index's physical size,
+        /// but for heavily updated tables a smaller fillfactor is better to minimize the need for page splits. The other index methods use fillfactor in different but roughly analogous ways;
+        /// the default fillfactor varies between methods.
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <param name="fillfactor">The fillfactor value from 10 to 100 can be selected</param>
+        /// <returns>The next step</returns>
+        public static ICreateIndexOptionsSyntax WithFillfactor(this ICreateIndexOptionsSyntax expression, int fillfactor)
+        {
+            var additionalFeatures = expression as ISupportAdditionalFeatures;
+            additionalFeatures.SetAdditionalFeature(IndexFillFactor, fillfactor);
+            return expression;
+        }
+
+        #region B-Tree
```

Comment:
I wonder if these can be refactored into sub-syntax so that you can say something like `.WithGin().FillFactor().Buffering()`. I am not that familiar with all these knobs in Postgres.  I mainly use the GIN for storing json data and that is it - what do you see as the common way people would write these migrations?

Reply:
@lillo42 Just making sure you saw this other comment too.